File: mns-reactor/src/main/java/cz/fb/manaus/reactor/categorizer/WinnerCountCategorizer.java
package cz.fb.manaus.reactor.categorizer;

import cz.fb.manaus.core.model.SettledBet;
import org.springframework.stereotype.Component;

@Component
public class WinnerCountCategorizer extends AbstractCountCategorizer {

    protected WinnerCountCategorizer() {
        super("winnerCount_", 6);
    }

    @Override
    protected int getCount(SettledBet bet) {
        return bet.getBetAction().getMarketPrices().getWinnerCount();
    }
}

File: mns-reactor/src/main/java/cz/fb/manaus/reactor/categorizer/WinnerCountCategorizer.kt
package cz.fb.manaus.reactor.categorizer

import cz.fb.manaus.core.model.SettledBet
import org.springframework.stereotype.Component

@Component
class WinnerCountCategorizer : AbstractCountCategorizer("winnerCount_", 6) {

    override fun getCount(bet: SettledBet): Int {
        return bet.betAction.marketPrices.winnerCount
    }
}
